src/floris/tools/floris_interface.py

```diff
@@ -71,30 +181,207 @@ def create_floris(self) -> None:
+            wake (:py:class:`~.wake.WakeModelManager` | None, optional): A container
+                class :py:class:`~.wake.WakeModelManager` with wake model information
+                used to calculate the flow field. Defaults to None.
+            layout_array (tuple[list[float]] | NDArrayFloat | None, optional): Array of
+                x- and y-locations of wind turbines, with shape (2 x N turbines).
+                Defaults to None.
+            turbine_id (list[str]| None, optional): The turbine mapping for each of the
+                turbines on the wind farm. This **must** be used if `layout_array` uses a
+                different number of turbines than the original model and this is a
+                multi turbine type farm.
+            wtg_id (list[str]| None, optional): The description for each of the
+                turbines on the wind farm. This should be used if there is a new number
+                of turbines in `layout_array`.
+            with_resolution (float, optional): Resolution of output flow_field. Defaults
+                to None.
+        """
+        reinitialize_farm = False
+        farm = {}
+        flow_field_dict = self.floris.flow_field._asdict()
+        if wind_speed is not None:
+            flow_field_dict["wind_speeds"] = wind_speed
+            reinitialize_farm = True
+        if wind_direction is not None:
+            flow_field_dict["wind_directions"] = wind_direction
+            reinitialize_farm = True
+        if wind_rose_probability is not None:
+            flow_field_dict["probability"] = wind_rose_probability
+        else:
+            flow_field_dict.pop("probability")
+        if wind_shear is not None:
+            flow_field_dict["wind_shear"] = wind_shear
+        if wind_veer is not None:
+            flow_field_dict["wind_veer"] = wind_veer
+        if specified_wind_height is not None:
+            flow_field_dict["reference_wind_height"] = specified_wind_height
+        if air_density is not None:
+            flow_field_dict["air_density"] = air_density
+        if wake is not None:
+            self.floris.wake = wake
+        if wind_layout:
+            pass  # TODO: will need for heterogeneous flow
+        if turbulence_intensity is not None:
+            pass  # TODO: this should be in the code, but maybe got skipped?
+        if turbulence_kinetic_energy is not None:
+            pass  # TODO: not needed until GCH
+        if with_resolution is not None:
+            # TODO: Update  whereever this goes
+            # TODO: This is the grid_resolution, so the xxGrid will have to be updated, except that grid_resolution is is Vec3, or int
+            # NOTE: probably not needed because this was for v2 Curl functionality, and we're adding a Curl solver
+            pass
+
+        self.floris.flow_field = FlowField.from_dict(flow_field_dict)
+        if reinitialize_farm:
+            farm = farm = self.floris.farm._asdict()
```

Comment:
Is the double-assign necessary?

Reply:
I think its a typo...

Reply:
Definitely a copypasta error. Thanks for catching this!